<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Vikram Chhatre" />
  <meta name="dcterms.date" content="2022-07-20" />
  <title>Markdown and RMarkdown</title>
  <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <style>
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="avenir-white.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Markdown and RMarkdown</h1>
<p class="author">Vikram Chhatre</p>
<p class="date">July 20, 2022</p>
</header>
<p><br><br></p>
<h2 id="table-of-contents">Table of Contents</h2>
<ul>
<li><p><a href="#software-install">1. Software Install</a></p>
<ul>
<li><p><a href="#install-rmarkdown">1.1 Install <code>RMarkdown</code></a></p></li>
<li><p><a href="#install-tinytex">1.2 Install <code>TinyTeX</code></a></p></li>
<li><p><a href="#install-pandoc-document-converter">1.3 Install <code>Pandoc</code></a></p></li>
</ul></li>
<li><p><a href="#data-for-todays-exercise">2. Data for today’s exercise</a></p></li>
<li><p><a href="#further-reading">3. Further Reading</a></p></li>
<li><p><a href="#creating-a-rmarkdown-document">4. Creating a RMarkdown Document</a></p></li>
</ul>
<p><br><br><br><br><br> <br><br><br><br><br> <br><br><br><br><br> <br><br><br><br><br> <br><br><br><br><br> <br><br><br><br><br> <br><br><br><br><br></p>
<h2 id="software-install">1. Software Install</h2>
<p>We will be using RStudio to create a RMarkdown document. But before we can do so, we will need to install following software:</p>
<p><br></p>
<h3 id="install-rmarkdown">1.1 Install <code>RMarkdown</code></h3>
<ul>
<li>Using CRAN repository (stable version, but could be slightly stale)</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;rmarkdown&quot;</span>)</span></code></pre></div>
<ul>
<li>Development version (cutting edge) from GitHub</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(devtools)</span>
<span id="cb2-2"><a href="#cb2-2"></a></span>
<span id="cb2-3"><a href="#cb2-3"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;rstudio/rmarkdown&quot;</span>)</span></code></pre></div>
<p><br></p>
<h3 id="install-tinytex">1.2 Install <code>TinyTeX</code></h3>
<p>RMarkdown is able to produce LaTeX formatted output (<code>.tex</code> format) but it requires <code>TinyTeX</code> a very small and easy to maintain TeX distribution. Installation is straightforward:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">install.packages</span>(<span class="st">&quot;tinytex&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2"></a></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="kw">library</span>(tinytex)</span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a>tinytex<span class="op">::</span><span class="kw">install_tinytex</span>()</span></code></pre></div>
<p><br></p>
<h3 id="install-pandoc-document-converter">1.3 Install <code>Pandoc</code> Document Converter</h3>
<p>Pandoc is a versatile document converter with numerous formats available including pdf and html. Install it from <a href="https://pandoc.org/installing.html">https://pandoc.org/installing.html</a>.</p>
<p><br><br></p>
<h2 id="data-for-todays-exercise">2. Data for today’s exercise</h2>
<p>This exercise uses a population genomic data set from study of two hybridizing <em>Populus</em> species. One of the goals of this study was to uncover signatures of natural selection in hybrid individuals. In order to do this, a genetic differentiation index called <code>Fst</code> is calculated between populations for two sets of genetic loci: those that have been determined to be under the effect of natural selection and those that are presumed neutral (i.e. no selection). We will be making a couple of figures from this data.</p>
<ul>
<li><p><code>sig.fst</code> contains data for SNPs under natural selection</p></li>
<li><p><code>neu.fst</code> contains data for SNPs presumed neutral</p></li>
<li><p><code>pval_counts.txt</code> contains data on bootstrapping</p></li>
</ul>
<p>If you run <code>git pull</code> in this repository’s clone directory on your computer, you should then see the <code>data</code> folder under this week’s section.</p>
<p><br><br></p>
<h2 id="further-reading">3. Further Reading</h2>
<ul>
<li><p><a href="https://bookdown.org/yihui/rmarkdown">R Markdown: A Definitive Guide</a> by Yihui Xie, J. J. Allaire, Garrett Grolemund</p></li>
<li><p><a href="https://kbroman.org/knitr_knutshell/pages/Rmarkdown.html">RMarkdown Tutorial by Karl Broman</a></p></li>
</ul>
<p><br><br></p>
<h2 id="creating-a-rmarkdown-document">4. Creating a RMarkdown Document</h2>
<p>If you were not in attendance for this session, follow the rest of this guide to create your document:</p>
<p><br></p>
<h3 id="create-document-preamble">4.1 Create Document Preamble</h3>
<p><br></p>
<h3 id="create-a-fst-density-plot">4.2 Create a Fst Density Plot</h3>
<p><strong>Note:</strong> Remember to enclose all R code between backticks in the following manner:</p>
<p><code>```{r} plot() ```</code></p>
<ul>
<li>Import libraries and data into R</li>
</ul>
<pre class="{r}"><code>library(ggplot2)
library(reshape2)

sig.fst &lt;- read.table(&quot;sig.fst&quot;, header=F)
neu.fst &lt;- read.table(&quot;neu.fst&quot;, header=F)
</code></pre>
<ul>
<li>Format and merge data</li>
</ul>
<pre class="{r}"><code>sig &lt;- data.frame(x=sig.fst$V2, label=rep(&quot;Significant (1764 SNPs)&quot;, 1764))

neu &lt;- data.frame(x=neu.fst$V2, label=rep(&quot;Neutral (1912 SNPs)&quot;, 1912))

fst &lt;- rbind(sig, neu)
</code></pre>
<ul>
<li>Make density plot</li>
</ul>
<pre class="{r}"><code>ggplot(fst, aes(x, y=..density.., fill=label)) + xlim(0,1) +
    geom_density(color=&quot;black&quot;, alpha=0.7) +
    scale_fill_manual(values=c(&quot;orange&quot;, &quot;lightgray&quot;)) +
    labs(x=expression(paste(&quot;Admixture Zone&quot;~italic(F)[ST])), y=&quot;Density (loci)&quot;,
        title=expression(paste(&quot;(b) &quot;~italic(F)[ST]~&quot; in &quot;~italic(&quot;P. angustifolia&quot;)~&quot; derived alleles&quot;))) +
    theme(legend.position=c(0.8,0.9), legend.title=element_blank())
</code></pre>
<p><br></p>
<h3 id="reproducibility-of-results">4.3 Reproducibility of results</h3>
<ul>
<li>Read the data file</li>
</ul>
<pre class="{r}"><code>pct &lt;- read.table(&quot;pval_counts.txt&quot;, header=T)</code></pre>
<ul>
<li>Assign colors to numbers</li>
</ul>
<pre class="{r}"><code>pct$fill &lt;- ifelse(pct$counts_pvalzero == 0, &quot;lightgray&quot;,
        ifelse(pct$counts_pvalzero == 100, &quot;orange&quot;, &quot;black&quot;))</code></pre>
<ul>
<li>Draw a histogram to show reproducibility</li>
</ul>
<pre class="{r}"><code>ggplot(pct, aes(counts_pvalzero, fill=fill)) +
    geom_histogram(binwidth=1) +
    scale_fill_manual(values=c(&quot;black&quot;, &quot;gray&quot;, &quot;orange&quot;),
        labels=c(&quot;Variabily Sig. (3847 SNPs)&quot;, &quot;Never Sig. (1912 SNPs)&quot;, &quot;Always Sig. (1764 SNPs)&quot;)) +
        labs(x=&quot;Num. of Introgress Replicates&quot;, y=&quot;Significantly introgressed loci&quot;,
            title=&quot;(a) Reproducibility of Introgress results&quot;) +
    theme(legend.position=c(0.8,0.9), legend.title=element_blank(), legend.text=element_text(size=8))
</code></pre>
<p><br><br></p>
<h3 id="compile-the-document">4.4 Compile the Document</h3>
<ul>
<li><p>If you are working within RStudio, simply click the <code>knit</code> button. This should generate an HTML file that RStudio will open up in a browser instance.</p></li>
<li><p>If you are working in a R terminal, follow these prompts:</p></li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">library</span>(RMarkdown)</span>
<span id="cb10-3"><a href="#cb10-3"></a></span>
<span id="cb10-4"><a href="#cb10-4"></a>rmarkdown<span class="op">::</span><span class="kw">render</span>(<span class="st">&quot;fst.Rmd&quot;</span>, <span class="st">&quot;html_document&quot;</span>)</span></code></pre></div>
<p><br><br></p>
<h3 id="example-output">4.5 Example Output</h3>
<ul>
<li>If your document was compiled correctly, you should see output like <a href="data/fst.html">this file</a>.</li>
</ul>
</body>
</html>
